File: WasteVisionWebBE/Domain/RoboflowModel/RoboflowModelMapper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DDDSample1.Domain.Roles;
using DDDSample1.Domain.Shared;

namespace DDDSample1.Domain.RoboflowModels
{
    public static class RoboflowModelMapper
    {
        public static RoboflowModelDto ToDto(RoboflowModel model)
        {
            if (model == null)
                return null;

            return new RoboflowModelDto(
                model.Id.AsGuid(),
                model.Description.AsString(),
                model.ApiKey.AsString(),
                model.ModelUrl.AsString(),
                model.LocalModelPath.AsString(),
                model.EndPoint.AsString(),
                model.Map.AsString(),
                model.Recall.AsString(),
                model.Precision.AsString(),
                model.Active
            );
        }

        public static RoboflowModel ToDomain(CreatingRoboflowModelDto dto)
        {
            if (dto == null)
                return null;

            return new RoboflowModel(
                new ApiKey(dto.ApiKey),
                new Url(dto.ModelUrl),
                new FilePath(dto.LocalModelPath),
                new Description(dto.Description),
                new Url(dto.EndPoint),
                new NumberDouble(dto.Map),
                new NumberDouble(dto.Recall),
                new NumberDouble(dto.Precision)
            );
        }

        public static List<RoboflowModelDto> ToDtoList(List<RoboflowModel> models)
        {
            if (models == null)
                return null;

            return models.Select(model => ToDto(model)).ToList();
        }

       
    }
}

File: WasteVisionWebFE/src/data/mappers/roboflowModel-mapper.ts
import { RoboflowModel } from "@/core/domain/RoboflowModel";
import {
  RoboflowModelDTO,
  CreateRoboflowModelDTO,
  UpdateRoboflowModelDTO,
} from "../dto/roboflowModel-dto";

export class RoboflowModelMapper {
  /**
   * Maps a RoboflowModelDTO to a domain RoboflowModel
   */
  public static toDomain(dto: RoboflowModelDTO): RoboflowModel {
    return {
      id: dto.id,
      description: dto.description,
      apiKey: dto.apiKey,
      modelUrl: dto.modelUrl,
      localModelPath: dto.localModelPath,
      endPoint: dto.endPoint,
      map: dto.map,
      recall: dto.recall,
      precision: dto.precision,
      active: dto.active,
    };
  }

  /**
   * Maps a domain RoboflowModel to a RoboflowModelDTO
   */
  public static toDTO(domain: RoboflowModel): RoboflowModelDTO {
    return {
      id: domain.id,
      description: domain.description,
      apiKey: domain.apiKey,
      modelUrl: domain.modelUrl,
      localModelPath: domain.localModelPath,
      endPoint: domain.endPoint,
      map: domain.map,
      recall: domain.recall,
      precision: domain.precision,
      active: domain.active,
    };
  }

  /**
   * Maps an array of RoboflowModelDTOs to domain RoboflowModels
   */
  public static toDomainList(dtos: RoboflowModelDTO[]): RoboflowModel[] {
    return dtos.map((dto) => this.toDomain(dto));
  }


}